app/renderer/js/main.ts

```diff
@@ -1040,6 +1042,24 @@ class ServerManagerView {
 			await this.openSettings('Network');
 		});
 	}
+
+	initLoginEventHandler() {
+		// This event gets fired whenever a webview triggers a basic authentication.
+		// We show an authentication dialog and return the username and password.
+		app.on('login', async (event, webContents, details, authInfo, callback) => {
+			// This doesn't seem to work, so it's done in app/main/index.ts as well.
+			event.preventDefault();
```

Comment:
The reason it doesn’t work is that you’re accessing `app` via `remote`, which doesn’t preserve synchronicity. We want to get rid of `remote` (#915). So assuming we need to do this in the renderer process, we should instead have the main process `login` handler contact the renderer process using IPC, probably using the same `makeRendererCallback` mechanism that the permission request handler uses.

Reply:
I've found PRs regarding replacing webview with BrowserView. That might make it easier/possible to attach the event listener directly to that as well. I tried it with the webview but didn't get it to work.

I have no experience with electron, so I probably missed something.